src/app/components/student-form/student-form.component.ts

```diff
@@ -20,16 +24,48 @@ export class StudentFormComponent implements OnInit {
     gender: new FormControl('')
   });
 
-  constructor(private studentService: StudentService) { }
+  public isEdit: boolean;
+
+  constructor(
+    private studentService: StudentService,
+    private route: ActivatedRoute,
+    private router: Router
+    ) {
+      this.route.params.subscribe(params => {
+        const studentId = params['id'];
+        if (studentId) {
+          this.getStudent(studentId);
+        }
+      });
+
+      this.router.events.pipe(
+        filter(event => event instanceof NavigationEnd)
+      ).subscribe((event: NavigationEnd) => {
+        this.isEdit = event.url.indexOf('edit') !== -1;
+      });
+    }
 
   ngOnInit() {
   }
 
-  add() {
-    const student = {...this.profileForm.value};
+  add(form: FormGroup): void {
+    const student: Student = {...form.value};
     this.studentService.createStudent(student)
       .subscribe();
 
-    this.profileForm.reset();
+    this.router.navigate(['/student']);
```

Comment:
This navigation throws an error: `Cannot match any routes. URL Segment: 'student'`